### ISSUE TITLE
Integrate Mirador with Collections

### ISSUE THREAD

**Issue by ColeDCrawford** [rsinghal/giza#90]
> Can we integrate Mirador with Collections such that the items in a Collection are viewable together as a Mirador manifest? 

This would require all items in the Collection to have IIIF resources, or a clear understanding of how to handle resources without IIIF resources. How would we handle items without images (eg videos)? Right now, only 20 sample items in ES have IIIF resources attached.

**Comment by npicardo** [comment]
MyGiza Collections were discussed in design meetings as necessarily relying on Mirador, so that dependency was expected.  Through this initial stage, video media wasn't expected to be in MyGiza Collections.  A separate section was designed for them on the Giza@School Lesson Topic pages, i.e. so they could be included in parallel with Collections, rather than as items in the Collections. Once the remaining issue with the Offloader App has been worked out, the tiny number of test IIIF images in the DB will jump to over 65,000.

**Comment by ColeDCrawford** [comment]
@lukehollis do you have those designs? Not seeing anything for this in the data model.

@npicardo - just curious, were there any designs for creating Collections? Right now that "+" link drops you into the Django admin interface. That's fine, but may be a bit jarring for end users who haven't seen that before.

**Comment by ColeDCrawford** [comment]
Following Nick's email - [http://dev.pixelsforhumans.com/giza/mygiza-collection.html#](http://dev.pixelsforhumans.com/giza/mygiza-collection.html) for designs.

I'm seeing a modal for "Add Collection" here which isn't implemented, but no further designs for how to actually add content there through a separate frontend workflow.
![giza-collections](https://user-images.githubusercontent.com/16374762/100758908-08d65880-33be-11eb-8f23-d6a5f8b7f5c9.gif)

**Comment by ColeDCrawford** [comment]
Designs for adding images to collections from an image page:
![01 - Artifact Page - Default](https://user-images.githubusercontent.com/16374762/100780035-3d0a4300-33d7-11eb-98c9-d804f8bbf912.png)
![02 - Artifact Page - Added to Collection](https://user-images.githubusercontent.com/16374762/100780041-3ed40680-33d7-11eb-84a0-1a8dbb0e401c.png)
![03 - Artifact Page - Added with Annotations](https://user-images.githubusercontent.com/16374762/100780048-41366080-33d7-11eb-8b1b-d128a7794ae6.png)

Thanks @npicardo. Were there any designs for adding content from the collection page, or just from an image page?

To be able to view a collection, yes, Mirador is needed. But this could have been built out before, as it's about adding content to the database managed by Django.

The annotations work is separate.

**Comment by ColeDCrawford** [comment]
@lukehollis do you need any other info for this? Or can this be moved from backlog to Groomed?

**Comment by ColeDCrawford** [comment]
This is related to #87. Right now, the only way an admin or an end user can add items to a collection is by filling in a list of comma-separated IDs in the Django admin interface.

The above designs outline the workflow for adding an item to a collection, as an end user, from the item page.

For an admin, there should be some sort of autocomplete feature to add an item to a collection from the Django admin Collection page. Ideally most of this work could be reused to provide a similar feature for a non-admin user on the frontend Collection page. I don't see a compelling reason that a non-admin user should ever see the Django admin interface.

**Comment by npicardo** [comment]
I agree, the dashboard should be admin only and that public users should not need to see it.  It would be impractical to require them to add images via the TMS mastermedia ID, which none will have the know-how to do. For admins using the dashboard, I personally have no problem with the current method of adding media for admins.  For public users, the designs always envisioned adding media from the front end via clicks associated with the media themselves.  The only other means Heather talked about was the possibility of an "add all images" kind of button on page views displaying multiple images (e.g. media search results?), but I haven't noticed that having made it through to final designs.  (Will look again to be sure.)

cheers,
Nick

________________________________

**Comment by ColeDCrawford** [comment]
Thanks for the clarification, @npicardo. That implementation on the admin side still seems a bit clunky to me, but it's your and Peter's call on how to prioritize tasks. If you have a chance to sort the "Groomed" column so the highest priority cards are at the top, that might help @lukehollis unless you're already communicating with him off-board on which tasks to tackle first.

Whether or not that button was in the designs doesn't really matter at this point because all the pending work for this and the other features is all backend / data model related.

Linking this to #94 as end users should have restricted permissions scoped to edited only their content.

**Comment by npicardo** [comment]
Thanks Cole.

I agree that it's a little clunky on the admin side.  In the interest of expedience, I feel as though I've no alternative but to weight the existence of basic functions over optimal convenience of those functions.

Apologies for about designs.  Lacking fine-grained knowledge of what does what on the back end, my main fallback is to understand the website in terms of front-end appearance and function.

I'll take a look at the "Groomed" tasks.  I'd though only the "To do" list was in need of re-ordering. On it!

cheers,
Nick

________________________________

**Comment by lukehollis** [comment]
Thanks for updating, @rsinghal--it sounds like Mirador is a separate issue than what's described with the admin side? Is this now a duplicate of https://github.com/rsinghal/giza/issues/87  ?

**Comment by ColeDCrawford** [comment]
#87 is specifically admin side, adding items to collections

I'll probably split these into separate tasks, but getting thoughts down here first.

- [ ] We need a way for end users to add items to collections, without visiting the Django admin. Probably a button on the item page as outlined above in the designs.
- [ ] We need correctly disambiguated items within a collection scope. I don't know what ID collections is currently searching on, but simply using a list of IDs is inadequate because the IDs are not namespaced. You end up with 3 IDs turning into 5 objects (eg 26830 gets both http://giza-web2.rc.fas.harvard.edu/photos/26830/full/ and http://giza-web2.rc.fas.harvard.edu/objects/26830/full/). See images
<img width="791" alt="Screen Shot 2020-12-11 at 4 15 20 PM" src="https://user-images.githubusercontent.com/16374762/101959864-3304f300-3bd4-11eb-998b-98f6a29ca416.png">
<img width="1161" alt="Screen Shot 2020-12-11 at 4 13 53 PM" src="https://user-images.githubusercontent.com/16374762/101959868-34ceb680-3bd4-11eb-8e0d-92fa62d9ea74.png">
- [ ] We need to know how mixed-content collections should display. What happens to a collection with a video, an image media record (manifest with 1 canvas), a Site (manifest with 1+ canvases), and another image/Site/Constituent/etc that does not yet have a DRS ID / manifest record? @npicardo above said video shouldn't be in collections, but there is currently nothing to stop a user from adding it there (see photo - I added ID 45695 and that adds both "Western Cemetary: Site: Giza" video and "Fragments of box coffin" object because they both have that ID). We would need to someone prevent people from adding videos in the first place. Also, how do we handle items that don't have DRS IDs yet?
<img width="1136" alt="Screen Shot 2020-12-11 at 5 30 42 PM" src="https://user-images.githubusercontent.com/16374762/101961060-ec64c800-3bd6-11eb-8118-bab659201896.png">
- [ ] Once we have a list of the content which can actually be displayed by Mirador, we will need to generate a IIIF Collection (an ordered list of manifests) for the Giza Collection's manifests. This needs to be generated on the fly or re-generated every time the collection is edited. I think as long as the Collection points to manifests rather than embedding them we would not need to regenerate if an item is edited.
- [ ] Once we have the above in place, then we can handle integrating Mirador. Adding Mirador is straightforward if all the data is there.
